### PR TITLE
First timer: Remove jQuery part 1

### LINKED ISSUE #538: Refactor code to eventually remove jquery dependency - part 1
## First Time?

This is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, consider leaving this one for someone new, and looking through our general [help wanted](https://github.com/publiclab/Leaflet.DistortableImage/issues) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

## Problem

**What happens:**

`JQuery` is the 2nd biggest resource we are pulling in (first is `Leaflet.js`), slowing down page loading time and we barely even use it.

## Solution

Let's start refactoring any jQuery find in the code base to vanilla JS. In this issue, we will tackle the below code. Please refactor the ajax  implementation to use XHR or fetch instead. 

https://github.com/publiclab/Leaflet.DistortableImage/blob/e0b51303151565dda66e075653c328f488acf169/src/edit/DistortableCollection.Edit.js#L260-L346

## Step by Step
- [ ] Claim this issue with a comment here, below, and ask any clarifying questions you need
- [ ] Fork the repository and set it up locally following the main repo README instructions [https://github.com/publiclab/Leaflet.DistortableImage](https://github.com/publiclab/Leaflet.DistortableImage)
 - [ ] Create a new feature branch with a unique name descriptive to the issue
- [ ] Try to fix the issue following the steps above, but even before you're done, you can:
  commit your changes to your branch and start a pull request (see [contributing to Public Lab software](https://publiclab.org/wiki/contributing-to-public-lab-software)) but mark it as "in progress" if you have questions or if you haven't finished
- [ ] Reference this issue in your pull request body
- [ ] Once you submit your pull request, if there's an additional checklist provided for getting it merged, get those boxes checked off. Either way, mention me @sashadev-sky for a review.

##  💬   Get help
If you need any help - here are some options: 

- Comment below
- Join our gitter chat at https://gitter.im/publiclab/publiclab

### PR DESCRIPTION
**Pull request by carlo-ev**
Fixes #538 

@sashadev-sky 
Thanks for the opportunity to contribute for the first time, please let me know if I have to change something.

Thank you once more!

### CODE REVIEW

File: src/edit/DistortableCollection.Edit.js [hunk 1]
```diff
@@ -303,28 +303,27 @@ L.DistortableCollection.Edit = L.Handler.extend({
...
         // repeatedly fetch the status.json
         self.updateInterval = setInterval(function intervalUpdater() {
-          $.ajax(statusUrl + '?' + Date.now(), {// bust cache with timestamp
-            type: 'GET',
-            crossDomain: true,
-          }).done(function(data) {
-            opts.updater(data);
-          });
+          var request = new XMLHttpRequest();
+          request.onload = function(event) {
```
Comment (sashadev-sky) [inline src/edit/DistortableCollection.Edit.js]: `.done` runs only on success, so to mirror that via xhr I think we should add some a conditional inside `onload` like `if (request.status === 200) { //... }`

Comment (carlo-ev) [reply src/edit/DistortableCollection.Edit.js:307]: Sorry for the long time to respond, things are shaky.
I missed the validation sorry, I have added the validation.

Comment (sashadev-sky) [reply src/edit/DistortableCollection.Edit.js:307]: no problem! I will take a look again now

File: src/edit/DistortableCollection.Edit.js [hunk 2]
```diff
@@ -303,28 +303,31 @@ L.DistortableCollection.Edit = L.Handler.extend({
...
```
Comment (sashadev-sky) [inline src/edit/DistortableCollection.Edit.js:307]: We can just remove the event parameter if we don't use it
```suggestion
          request.onload = function() {
```

File: src/edit/DistortableCollection.Edit.js [hunk 3]
```diff
@@ -303,28 +303,31 @@ L.DistortableCollection.Edit = L.Handler.extend({
...
+            if (request.status === 200) {
+              opts.updater(JSON.parse(this.response));
+            }
+          };
+          request.open('GET', statusUrl + '?' + Date.now());
```
Comment (sashadev-sky) [inline src/edit/DistortableCollection.Edit.js:312]: We're going to want to copy over the `crossDomain: true` attributes from the jQuery as well! Especially important for the `POST` request.

Comment (carlo-ev) [reply src/edit/DistortableCollection.Edit.js:312]: I thought the `crossDomain: true` option was only available on Ajax, I'm changing the resquests to the Request API anyways to keep them consistent with the rest. Just asking for further knowledge.

Comment (sashadev-sky) [reply src/edit/DistortableCollection.Edit.js]: yes you are right I was referring to the plain vanilla version of allowing cross domain requests. Great I will review that today :) thank you!!

**Comment by sashadev-sky** [comment]
@carlo-ev I wanted to rebase for you but i've noticed lately i don't have permissions to push to people's PR's  automatically anymore.. would you want to add me as a contributor to your fork so i can do that for you? https://stackoverflow.com/questions/14969281/git-how-to-give-people-push-access-that-needs-to-be-merged-in-by-organizers#14972607

or you can just rebase yourself

**Comment by sashadev-sky** [comment]
@carlo-ev **note before you make any changes you'l want to pul in locally the rebase i did for you here. -- so `$ git pull https://github.com/publiclab/Leaflet.DistortableImage.git first-timer-jq-pt1`**

**Comment by carlo-ev** [comment]
I fix the "opts" mistype and changed both response return from json to text.
Sorry for wasting your time with those mistakes.
I hope this one is good.

**Comment by sashadev-sky** [comment]
@carlo-ev no worries, just make sure you are checking your work in the future by running the code locally. Rebasing your dist file and then reviewing again!

**Comment by sashadev-sky** [comment]
@jywarren @VladimirMikulic this looks good to me, any idea why it's getting stuck on the "compositing" step again? I think in the past that has indicated a CORS issue, but i'm not getting any error messages in the console and the response headers look good to me? I have it up on my gh-pages here: https://sasha.mapknitter.org/examples/select.html

File: src/edit/DistortableCollection.Edit.js [hunk 4]
```diff
@@ -303,28 +303,28 @@ L.DistortableCollection.Edit = L.Handler.extend({
...
         self.updateInterval = setInterval(function intervalUpdater() {
-          $.ajax(statusUrl + '?' + Date.now(), {// bust cache with timestamp
-            type: 'GET',
-            crossDomain: true,
-          }).done(function(data) {
-            opts.updater(data);
-          });
+          var request = new Request(statusUrl+'?'+Date.now(), {method: 'GET'});
+          fetch(request).then(function(response) {
+            if (response.status === 200) {
```
Comment (VladimirMikulic) [inline src/edit/DistortableCollection.Edit.js:308]: Hard coding this to 200 is not good. What if later it changes to 202, which is also valid code, just like any 200 status code. This should be replaced with `response.ok`. [Read More](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok)

File: src/edit/DistortableCollection.Edit.js [hunk 5]
```diff
@@ -303,28 +303,28 @@ L.DistortableCollection.Edit = L.Handler.extend({
...
-          success: function(data) { opts.handleStatusRes(data); }, // this handles the initial response
-        });
+        var form = new FormData();
+        form.append('collection', JSON.stringify(opts.collection));
+        form.append('scale', opts.scale);
+        form.append('upload', true);
+        var requestOptions = {method: 'POST', body: form};
+        var request = new Request(opts.exportStartUrl, requestOptions);
+        fetch(request).then(function(response) {
+          if (response.status === 200) {
```
Comment (VladimirMikulic) [inline src/edit/DistortableCollection.Edit.js:324]: Here as well.

**Comment by VladimirMikulic** [comment]
@sashadev-sky could you clarify, where is it stuck? What action should be triggered?

**Comment by sashadev-sky** [comment]
@VladimirMikulic i'm thinking since this was meant as an FTO and it's already implemented without async/await, we can save that for another PR. Are you on board with that? sorry should've gotten you involved earlier!

Also did you try it out on the page i linked? It just continues compositing endlessly. Meaning something is going wrong but it's not being indicated in the console.

**Comment by sashadev-sky** [comment]
by that I mean I see the message `{status_url: "https://mapknitter-exports-warps.storage.googleapis.com/1587431619/status.json", status: "compositing", tms: false, geotiff: "public/warps/1587431619/1587431619.tif", zip: false, …}`

**Comment by sashadev-sky** [comment]
@VladimirMikulic sorry that was my bad!! I turned on enforce https in my gh-pages settings. I'll make the change you requested because that was actually my suggestion not the contributors and request your review again for merge

**Comment by VladimirMikulic** [comment]
That's what I suspected haha :)

**Comment by sashadev-sky** [comment]
@VladimirMikulic I'll make a follow up FTO!

**Comment by sashadev-sky** [comment]
@carlo-ev thank you so much for the great work!

**Comment by carlo-ev** [comment]
@sashadev-sky thank you so much for the opportunity to help! Hope there's some more minor issues I can help with.

**Comment by sashadev-sky** [comment]
@carlo-ev what are you interested in?

**Comment by carlo-ev** [comment]
@sashadev-sky maybe issues #338 or #311 if that's ok, saw a lot already claimed on the "help wanted".

**Comment by sashadev-sky** [comment]
Yes, nobody is currently working on those so that would be awesome!! Thank you :) Feel free to let me know if you need any guidance on them! @carlo-ev

**Pull request merged**